test/jdk/com/sun/net/httpserver/SelCacheTest.java

```diff
@@ -74,7 +77,7 @@ public static void main(String[] args) throws Exception {
             }
             s2 = HttpsServer.create(addr, 0);
             // Assert that both files share the same parent and can be served from the same `FileServerHandler`
-            assert smallFilePath.getParent().equals(largeFilePath.getParent());
+            assertEquals(smallFilePath.getParent(), largeFilePath.getParent());
```

Comment:
That's OK. You could have kept `assert` here as it should not happen and would point to an issue with the logic in the test. But assertEquals will show us both paths if that happens, which is a +.